src/AlgebraicGeometry/TropicalGeometry/groebner_fan.jl

```diff
@@ -0,0 +1,544 @@
+    end
+    return cone_from_inequalities(matrix(QQ,inequalities))
+end
+
+
+
+# Computes the unique primitive vector that lies on the sum of all rays and is orthogonal to the lineality space.
+# If the cone has no rays, returns the zero vector.
+# Serves as a unique identifier for cones in a polyhedral fan that is compatible with respect to symmetries which permute the coordinates.
+function unique_identifying_point(C::Cone)
+
+    # compute rays modulo lineality space
+    R,_ = rays_modulo_lineality(C)
+
+    # if there are none, return zero vector
+    if isempty(R)
+        return zeros(fmpz,ambient_dim(C))
+    end
+    # otherwise compute the sum of rays, cast it to a rational vector
+    # and return a primitive integer vector
+    pt = Vector{fmpq}(sum(R))
+    return numerator.(pt .* lcm(denominator.(pt)))
+
+end
+
+
+
+# Returns an interior points on each facet.
+# We use unique_identifying_points because we also use them to track which facets have been traversed
+function interior_facet_points(C::Cone)
+    return unique_identifying_point.(faces(C,dim(C)-1))
+end
+
+
+
+# Returns a primitive outer normal vector for each facet.
+function outer_normal_vectors(C::Cone)
+    outerNormalMatrix = linear_inequality_matrix(facets(C))
+    normals = Vector{Vector{fmpz}}()
+    for i in 1:nrows(outerNormalMatrix)
+        v = [outerNormalMatrix[i,:]...]
+        push!(normals,numerator.(v .* lcm(denominator.(v))))
+    end
+    return normals
+end
+
+
+
+# Returns the initial form of polynomial g with respect to weight vector u.
+# Requires a monomial ordering that is compatible with respect to u, i.e., the leading monomial is of highest weighted degree.
+function initial(g::MPolyElem, ordering::MonomialOrdering, u::Vector{fmpz})
+
+    lt,tail = Iterators.peel(terms(g,ordering=ordering));
+    d = dot(u,fmpz.(leading_exponent_vector(lt)))
+
+    initial_terms = [s for s in tail if dot(u,fmpz.(leading_exponent_vector(s)))==d]
+    push!(initial_terms,lt)
+    return sum(initial_terms)
+
+end
+# Applies the function above to each polynomial in a list of polynomials.
+function initial(G::Vector{<:MPolyElem}, ord::MonomialOrdering, u::Vector{fmpz})
+    return initial.(G,Ref(ord),Ref(u))
+end
+
+
+
+# Assumes:
+# - G is a Groebner basis with respect to ord
+# Returns:
+# - a reduced Groebner basis with respect to ord
+function interreduce(G::Vector{<:MPolyElem}, ord::MonomialOrdering)
+
+    # sort G by its leading monomials (result is smallest first)
+    # then reduce G[i] by G[1:i-1] for i=length(G),...,2
+    sort!(G,
+          by=g->leading_monomial(g,ordering=ord),
+          lt=(f,g)->(cmp(ord,f,g)<0))
+    for i in 2:length(G)
+        G[i] = reduce(G[i],G[1:i-1],ordering=ord)
+    end
+    return G
+
+end
+
+
+
+# Assumes:
+# - Hnew is a Groebner basis of the initial ideal w.r.t. ordH
+# - G is a Groebner basis of the original ideal w.r.t. ordG
+# - ordH and ordG are adjacent orderings
```

Comment:
Is this tested in `groebner_lift`?

Reply:
No, it is not.  I left it out because it requires some non-trivial computations and it is not necessary.

To test the adjacency under the assumptions that `Hnew` and `G` are Groebner bases w.r.t. `ordH` and `ordG`, respectively, one would need to verify
1. that there is a weight vector `w` with `initial_w(G)=Hnew` (as sets of polynomials),
2. `lead(G,ordG)` and `lead(G,ordH)` are not the same (as sets of polynomials).

Point 1 implies that `ordG` and `ordH` are either adjacent or the same, Point 2 implies that they are not the same. (Note that "adjacent" here means that the Groebner cones share a common face containing `w`, but that face need not be one-codimensional)